packages/ra-core/src/i18n/Translate.spec.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import {
+    Args,
+    Basic,
+    NoTranslation,
+    NoTranslationWithChildren,
+} from './Translate.stories';
+import { TestTranslationProvider } from './TestTranslationProvider';
+import { Translate } from './Translate';
+
+describe('<Translate />', () => {
+    it('should render the translation', () => {
+        const { container } = render(<Basic />);
+        expect(container.innerHTML).toBe('My Translated Key');
+    });
+
+    it('should render the translation event if children is set', () => {
+        const { container } = render(
+            <TestTranslationProvider
```

Comment:
Instead of copying the code from the stories, you'd better render the stories themselves.